src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -5,17 +5,60 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import codeu.model.store.basic.UserStore;
+import codeu.model.data.User;
+import java.time.Instant;
+import java.util.UUID;
 
-    /**
-     * Servlet class responsible for user registration.
-     */
-    public class RegisterServlet extends HttpServlet {
+/**
+* Servlet class responsible for user registration.
+*/
+public class RegisterServlet extends HttpServlet {
+
+ @Override
+ public void doGet(HttpServletRequest request, HttpServletResponse response)
+     throws IOException, ServletException {
+
+   request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+ }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+        throws IOException, ServletException {
 
-        @Override
-        public void doGet(HttpServletRequest request, HttpServletResponse response)
-                throws IOException, ServletException {
+    String username = request.getParameter("username");
+    String password = request.getParameter("password");
 
-            response.getWriter().println("<h1>RegisterServlet GET request.</h1>");
-        }
+    if (!username.matches("[\\w*\\s*]*")) {
+        request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
+        request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+        return;
+    }
+
+    if (userStore.isUserRegistered(username)) {
+        request.setAttribute("error", "That username is already taken.");
+        request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+        return;
+    }
+
+    User user = new User(UUID.randomUUID(), username, password, Instant.now());
+    userStore.addUser(user);
+
+    response.sendRedirect("/login");
+    }
+    private UserStore userStore;
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setUserStore(UserStore.getInstance());
```

Comment:
Perhaps instead the constructor could take a UserStore? That way it isn't something that an external caller can change through setUserStore.